lib/engine-object-file/src/artifact.rs

```diff
@@ -488,6 +506,11 @@ impl Artifact for ObjectFileArtifact {
 
     /// Serialize a ObjectFileArtifact
     fn serialize(&self) -> Result<Vec<u8>, SerializeError> {
-        Ok(self.module_bytes.clone())
+        unimplemented!();
+        // let mut builder = ar::Builder::new(Vec::new());
+        // let header = Header::new(format!(b"baz.txt").to_vec(), file_size)
+        // builder.append(&header, file_content).unwrap();
+
+        // Ok(self.module_bytes.clone())
```

Comment:
The idea here is to create an archive file with all the object files. I'll wait to see @nlewycky thoughts on this.
I think llvm-ld can link properly archive files (also with `--whole-archive` option, if needed), so I think this would solve the limit file issue when Linking, as well as the CMD call size < 128Kb

Reply:
In the future we should link `lld` in directly instead of running it as a subprocess which will also let us bypass any file descriptor and command line limits by simply not using command lines or files.